### PR TITLE
reusable workflow for linux integrations

### PR DESCRIPTION
**Pull request by davidgamero**
# Description

Remove the existing shell script generated integration testing configuration in favor of using reusable github actions and a matrix

Fixes # (issue)
Feature # (details)

## Type of change

- [x] Refactor

## How Has This Been Tested?

- [x] Unit Test:
- [x] E2E Test:
- [ ] Other Test:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/integration-per-language.yml [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+          --variable BUILDERVERSION=1.11 \
+          --variable SERVICEPORT=8080 \
+          --variable NAMESPACE=testNamespace \
+          --variable IMAGENAME=testImage \
+          --variable IMAGETAG=latest \
+          --variable ENTRYPOINT=myapp.py
+      - name: Validate JSON
+        run: |
+          npm install -g ajv-cli@5.0.0
+          ajv validate -s test/dry_run_schema.json -d test/temp/dry-run.json
```
Comment (Tatsinnit) [inline .github/workflows/integration-per-language.yml:58]: ☕️ minor question: we are using ajv as json validation for specific preferred reasons or something else? (Asking only for archival reasons please)

File: .github/workflows/integration-per-language.yml [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+        ports:
+          - 5001:5000
+    needs: helm-dry-run
+    env:
+      imagename: registry:5001/testapp
+      serviceport: 80
+      ingress_test_args: "-a webapp_routing --variable ingress-tls-cert-keyvault-uri=test.cert.keyvault.uri --variable ingress-use-osm-mtls=true --variable ingress-host=host1"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
```
Comment (Tatsinnit) [inline .github/workflows/integration-per-language.yml:73]: ☕️💰 possible update to “v4” here as well for checkout option.

File: .github/workflows/integration-per-language.yml [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+          rm -rf ./langtest/overlays
+      - run: ./draft -v create -c ./test/integration/${{inputs.language}}/helm.yaml -d ./langtest/ --skip-file-detection
+      - name: start minikube
+        id: minikube
+        uses: medyagh/setup-minikube@master
+        with:
+          insecure-registry: "host.minikube.internal:5001,10.0.0.0/24"
+      # Runs Helm to create manifest files
+      - name: Bake deployment
+        uses: azure/k8s-bake@v3.0.1
```
Comment (Tatsinnit) [inline .github/workflows/integration-per-language.yml:96]: ☕️ do we want to pin this to a hash?

File: .github/workflows/integration-per-language.yml [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+      - name: Fail if any error
+        if: steps.deploy.outcome != 'success'
+        run: |
+          kubectl get po
+          echo "Deployment failed, check above logs and previous steps to isolate the issue"
+          exit 6
+  kustomize-dry-run:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (Tatsinnit) [inline .github/workflows/integration-per-language.yml:194]: ☕️ same: minor note please: possible candidate for “v4”

**Review by Tatsinnit — APPROVED** [review]
❤️🙏 Thank you for the kind ping! The PR is looking great; I’ve added a few thoughts. It might also be worth checking if we can upgrade checkout to v4. Additionally, I noticed that download-artifact is tied to a specific hash, while for others, we only use the version—just an observation to share.

Everything else looks good, and I trust this has been tested against a fork or similar setup.

Thanks so much!

File: .github/workflows/integration-per-language.yml (hunk 3 above)
Comment (davidgamero) [reply .github/workflows/integration-per-language.yml:96]: we can use semver when we migrate thesee to [immutable actions](https://github.com/features/preview/immutable-actions)

**Comment by davidgamero** [comment]
once the source of truth for integration tests is moved into a reusable workflow, dependabot will be able to open PRs on theese as well which will resolve many of the version bumps.

previously dependabot couldn't upgrade the actions as they were re-generated from the bash file each time

File: .github/workflows/integration-per-language.yml (hunk 1 above)
Comment (davidgamero) [reply .github/workflows/integration-per-language.yml:58]: there is no preference for ajv outside that it allows validating the schema of our json outputs.

happy to accept PRs for an alternative that better fits the needs of the project

**Review by Tatsinnit — APPROVED** [review]

**Review by Tatsinnit — APPROVED** [review]

**Review by jaiveerk — APPROVED** [review]

**Pull request merged**